### DISCUSSION TITLE
Importing CSS when use some component

### DISCUSSION THREAD

**Discussion by luizribeiro-hotmart** [webpack/webpack#15058] [Q&A] [answered]
Hey guys!

I have created an external package (based on React without CRA) and I'm using it inside another project based on React (without CRA either).

I'm using mini-css-extract-plugin in the external package to minimize the bundle, but because of the extract I need to always import the CSS when I use this external package.

Is that a way to, when I import a component of this external package, import automatically its CSS?

example:

I have a button component in this external package. When I build the package, I got this:

<img width="449" alt="Captura de Tela 2021-12-27 às 08 48 42" src="https://user-images.githubusercontent.com/77842041/147469061-325ed9c1-b042-4dc5-9743-730f1cce1f08.png">

So to use it I need to import like this:

<img width="760" alt="Captura de Tela 2021-12-27 às 08 49 46" src="https://user-images.githubusercontent.com/77842041/147469161-39665c39-4c17-4de6-92ca-791f3df6099c.png">

I'd like to use the CSS without imort it (like html-template-plugin), but without use it, because the external package is a design system, so it doesn't export HTML templates, just js components.

Is it possible?

I have others CSS variations too, like button small, large, warning, info etc. It'll be just awesome to use a prop on the component and it's automatically import the respective CSS!

**Comment by luizribeiro-hotmart** [comment, marked as answer]
can anyone help me??

**Reply by alexander-akait** [reply]
I think better ask it on stackoverflow, because it is an architecture problem, you faced with it using any bundler, JS is not support CSS import (currently). When you do library I recommend avoid using CSS imports, just do default CSS and provide instruction in docs how to use them. Some developers use CSS-in-JS, some can want to replace your styles, some extend them. 

> I have others CSS variations too, like button small, large, warning, info etc. It'll be just awesome to use a prop on the component and it's automatically import the respective CSS!

You need write this logic on component side...

**Reply by luizribeiro-hotmart** [reply]
I see... thanks for your answer!
